### PR TITLE
Add project list endpoint

### PR DESCRIPTION
**Pull request by dcramer**
This is primarily catering to upcoming internal admin tools.

@getsentry/api

### CODE REVIEW

File: src/sentry/api/endpoints/project_index.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        if request.auth:
+            if hasattr(request.auth, 'project'):
+                queryset = queryset.filter(
+                    id=request.auth.project_id,
+                )
+            elif request.auth.organization is not None:
+                queryset = queryset.filter(
+                    organization=request.auth.organization.id,
+                )
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_index.py]: Isn't it possible for this to fall back to the "else" here, resulting in a queryset that returns ALL projects? I think we should either assert hard here, or do like, `queryset = queryset.none()`.

Comment (dcramer) [inline src/sentry/api/endpoints/project_index.py]: yes its possible -- we dont validate many of these parameters and you *should* be able to return all projects.

Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_index.py]: I agree that you should, but that case should always be guarded with a superuser check. In this path, that's not checked.

Comment (dcramer) [inline src/sentry/api/endpoints/project_index.py]: why should only superusers be allowed to return projects are being deleted?

Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_index.py]: What does this have to do with projects being deleted? That's a completely different/unrelated check.

Comment (dcramer) [inline src/sentry/api/endpoints/project_index.py]: oh misread emails and thought this was related to status

the "else" is "I am a superuser so I get all projects"

Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_index.py]: So it's not worth guarding explicitly against superuser? I'm not familiar with the auth stuff enough to say `hasattr(request.auth, 'project') and request.auth.organization is not None == superuser`, and if so, that seems fragile to me since it's relying on an implementation detail.

File: src/sentry/api/endpoints/project_index.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            else:
+                queryset = queryset.none()
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_index.py]: :ok_hand:

**Comment by mattrobenolt** [comment]
:beers: :beers:
